backend/compound_library/src/resolvers/compound_instances_res.rs

```diff
@@ -0,0 +1,76 @@
+// src/resolvers/compound_instances_res.rs
+
+use crate::entities::{compound_instances, compound_types};
+use async_graphql::{ComplexObject, Context, Object};
+use opa_client::subject_authorization;
+use sea_orm::{ActiveValue, DatabaseConnection, EntityTrait, ModelTrait};
+use the_paginator::graphql::{CursorInput, ModelConnection};
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Default)]
+pub struct CompoundInstanceQuery;
+
+#[derive(Debug, Clone, Default)]
+pub struct CompoundInstanceMutation;
+
+#[ComplexObject]
+impl compound_instances::Model {
+    async fn compound_types(
+        &self,
+        ctx: &Context<'_>,
+    ) -> async_graphql::Result<Vec<compound_types::Model>> {
+        subject_authorization!("xchemlab.compound_library.read_compound", ctx).await?;
+        let db = ctx.data::<DatabaseConnection>()?;
+        Ok(self.find_related(compound_types::Entity).all(db).await?)
+    }
+}
+
+#[Object]
+impl CompoundInstanceQuery {
+    async fn compound_instances(
+        &self,
+        ctx: &Context<'_>,
+        cursor: CursorInput,
+    ) -> async_graphql::Result<ModelConnection<compound_instances::Model>> {
```

Comment:
Do we want to be able to filter this?

Reply:
we can already filter the pages right? or did you mean something else?

Reply:
Yeah, like filtering by plate, or well number. Probably not needed now, but worth considering how we'd do it